rust/src/lib/nispor/ethernet.rs

```diff
@@ -18,6 +19,26 @@ fn gen_eth_conf(np_iface: &nispor::Iface) -> EthernetConfig {
     if let Some(sriov_info) = &np_iface.sriov {
         eth_conf.sr_iov = Some(gen_sriov_conf(sriov_info));
     }
+    if let Some(sr_iov) = &mut eth_conf.sr_iov {
+        let p: path::PathBuf = [
+            "/sys/class/net",
+            &np_iface.name,
+            "device/sriov_drivers_autoprobe",
+        ]
+        .iter()
+        .collect();
+        if let Ok(contents) = fs::read_to_string(&p) {
+            match contents.as_str().parse::<u8>() {
+                Ok(i) => {
+                    sr_iov.drivers_autoprobe =
+                        if i == 0 { Some(false) } else { Some(true) };
+                }
+                Err(err) => {
+                    log::warn!("failed to parse {:?}: {:?}", &p, err);
```

Comment:
Do not use debug out, use `PathBuf::display()` and the error as Display trait implemented.

Reply:
I have tested this and it works.

```
[2024-04-05T20:31:32Z WARN  nmstate::nispor::ethernet] failed to parse /sys/class/net/eno12409/device/sriov_drivers_autoprobe: ParseIntError { kind: InvalidDigit }
```